client/my-sites/stats/stats-subscribers-overview/index.tsx

```diff
@@ -1,57 +1,90 @@
 import { CountComparisonCard } from '@automattic/components';
+import { UseQueryResult } from '@tanstack/react-query';
 import { translate } from 'i18n-calypso';
 import React from 'react';
+import useSubscribersQuery from 'calypso/my-sites/stats/hooks/use-subscribers-query';
 
-function SubscribersOverviewCardStats() {
-	const overviewCardStats = [
-		{
-			heading: translate( 'Today' ),
-			count: 1410,
-		},
-		{
-			heading: translate( '30 days ago' ),
-			count: 1080,
-			previousCount: 1080,
-		},
-		{
-			heading: translate( '60 days ago' ),
-			count: 1000,
-			previousCount: 1080,
-		},
-		{
-			heading: translate( '90 days ago' ),
-			count: 970,
-			previousCount: 1080,
-		},
-	];
-	return overviewCardStats;
+interface SubscribersData {
+	period: string;
+	subscribers: number;
+	subscribers_change: number;
 }
 
-function SubscribersOverviewCards() {
-	const overviewCardStats = SubscribersOverviewCardStats();
+interface SubscribersDataResult {
+	data: SubscribersData[];
+	unit: string;
+	date: string;
+}
 
-	return (
-		<div className="overview-cards-list">
-			{ overviewCardStats.map( ( overviewCardStat ) => (
-				<CountComparisonCard
-					key={ overviewCardStat.heading }
-					heading={ overviewCardStat.heading }
-					count={ overviewCardStat.count }
-					previousCount={ overviewCardStat.previousCount }
-					showValueTooltip
-					icon={ false }
-				/>
-			) ) }
-		</div>
-	);
+interface SubscribersOverviewProps {
+	siteId: number;
+}
+
+function extractCountsAtIndexes( subscribersData: SubscribersData[], indexes: number[] ): number[] {
+	return indexes.map( ( index ) => subscribersData[ index ]?.subscribers || 0 );
+}
+
+// all comparisons are being compared to todays's count to show growth up to today
+function SubscribersOverviewCardStats( subscribersData: number[] ) {
+	const daysToDisplay = [ 0, 30, 60, 90 ];
+	const overviewCardStats: {
+		heading: string;
+		count: number;
+		previousCount?: number;
+	}[] = [];
+
+	daysToDisplay.forEach( ( day, index ) => {
```

Comment:
This logic works, and it would be great if we had a significant number of modules.
With four components, I prefer the previous approach with an object with all 4 of them written down instead of a loop.
This is a personal preference, so you can go with this function, but I think the other one is more readable.

Reply:
I agree with you here in regards to readability. I'm working on the four component approach today and think I will simplify things back to things without the loop.